client/www/scripts/modules/arc/arc.module.js

```diff
@@ -148,24 +163,41 @@ Arc.run([
 
       // Redirect to login if route requires auth and you're not logged in
       $rootScope.$on('$stateChangeStart', function (event, next) {
+        function isAppModule(url){
+          return LicensesService.getArcFeatures()
+            .then(function(arcFeatures){
+              return _.contains(arcFeatures, url.substr(1)); //remove '/' off page url
+            });
+        }
 
-        if ( !ArcUserService.isAuthUser() && next.url !== '/login' ) {
-          event.preventDefault(); //prevent current page from loading
-          $state.go('login');
-        } else {
-          //fire off segment.io identify from cookie values
-          segmentio.identify(ArcUserService.getCurrentUserId(), {
-            name : ArcUserService.getCurrentUsername(),
-            email : ArcUserService.getCurrentUserEmail()
-          });
-          //fire off segment.io event on module invocation and ignore home, login, register
-          if (!_.contains(CONST.NON_ARC_MODULES, next.name)) {
-            segmentio.track(next.name);
+        function handleStateChange(){
+          if ( !ArcUserService.isAuthUser() && next.url.indexOf('/login') === -1 ) {
+            event.preventDefault(); //prevent current page from loading
+            $state.go('login', { ref: next.name });
+          } else {
+            //fire off segment.io identify from cookie values
+            segmentio.identify(ArcUserService.getCurrentUserId(), {
+              name : ArcUserService.getCurrentUsername(),
```

Comment:
this was already there. I think it should be getCurrentUsername because everything it references is using `Username` not `UserName`.